perllib/FixMyStreet/App/Controller/Report.pm

```diff
@@ -449,20 +453,23 @@ sub inspect : Private {
                 $redirect_uri = $c->uri_for( $problem->url );
             }
 
-            # Or if inspector, redirect back to /around at this report's
-            # location with the right filters. We go here rather than the
-            # shortlist because it makes it much simpler to inspect many reports
-            # in the same location. The shortlist is always a single click away,
-            # being on the main nav.
+            # Or if inspector, redirect back to the map view they came from
+            # with the right filters. If that wasn't set, go to /around at this
+            # report's location.
+            # We go here rather than the shortlist because it makes it much
+            # simpler to inspect many reports in the same location. The
+            # shortlist is always a single click away, being on the main nav.
             if ($c->user->has_body_permission_to('planned_reports')) {
```

Comment:
As this overrides the work done before, should it perhaps come first in the previous if statement (I say first so that it doesn't have to bother doing the `owns_problem` check if this one passes).